openpype/hosts/maya/plugins/publish/validate_model_name.py

```diff
@@ -71,15 +71,15 @@ def is_group(group_name):
             invalid.append(top_group)
         else:
             if "asset" in r.groupindex:
-                if m.group("asset") != legacy_io.Session["AVALON_ASSET"]:
+                if m.group("asset") != get_current_asset_name():
                     cls.log.error("Invalid asset name in top level group.")
                     return top_group
             if "subset" in r.groupindex:
                 if m.group("subset") != instance.data.get("subset"):
                     cls.log.error("Invalid subset name in top level group.")
                     return top_group
             if "project" in r.groupindex:
-                if m.group("project") != legacy_io.Session["AVALON_PROJECT"]:
+                if m.group("project") != get_current_project_name():
```

Comment:
These could be taken from `instance.context.data`
Or even better, **they should actually be taking the asset from the instance!**

Reply:
I'm not sure if instances from legacy creators have `"asset"` key set. Made it compatible for both with TODO.